spec/numberformat.html

```diff
@@ -120,17 +129,21 @@ <h1>
           1. Let _hasFd_ be *true*.
         1. Else,
           1. Let _hasFd_ be *false*.
-        1. Let _needSd_ be _hasSd_.
-        1. If _hasSd_ is *true*, or _hasFd_ is *false* and _notation_ is *"compact"*, then
-          1. Let _needFd_ be *false*.
-        1. Else,
-          1. Let _needFd_ be *true*.
+        1. Let _needSd_ be *true*.
```

Comment:
Full disclosure: this entire `needs*d`/`has*d` part of the algorithm always confuses me so for the time being I'd trust @gibson042 to review it.